internal/tokenstatus/token_status.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2024 Stacklok, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package tokenstatus provides token status functions
+package tokenstatus
+
+import (
+	"sync"
+
+	"github.com/stacklok/minder/internal/db"
+)
+
+// Store stores the rate limit status of token
+type Store struct {
```

Comment:
I don't have a great use-case in mind, so this might be an over engineering suggestion, but since the members of store are not exported anyway, I wonder if creating an interface and exposing it would allow for easier testing.

Reply:
I am ok with doing so, but as you said, I also don't have a use case for that. As far as testing is concerned, I think the current (main) implementation is fine, and we wouldn't need any method, unless we want to do something similar to mocking, counting method calls, which I feel is not required.

Reply:
OK, let's leave it like this.